### PR TITLE
refactor(app): add protocol run details run log

### LINKED ISSUE #8753: Run tab - Run log
# Overview
The protocol run details page has a run tab with 2 sections. The second section on the run tab is the protocol run log.

# Design
- The run log before the protocol run is started: https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/%5Bworking%5D-Milestone-1%3A-Read-and-Execute?node-id=1297%3A101653
- The run log when the protocol is in progress/running: https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/%5Bworking%5D-Milestone-1%3A-Read-and-Execute?node-id=1297%3A101653
- The run log when the protocol run is complete: https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/%5Bworking%5D-Milestone-1%3A-Read-and-Execute?node-id=1297%3A101653

# Acceptance Criteria

- [ ] Future steps should have an unadorned gray styling 

- [ ] The currently running step should have a blue border and background and a "Current Step" title

- [ ] Completed steps are styled differently with start and end times for each step and a green background

- [ ] Clicking on "Download run log" should prompt the user to download the run log file locally

# Selectors for E2E

- [ ] Run log tab
- [ ] download run log 
- [ ] jump to current step
- [ ] Number of steps

### PR DESCRIPTION
**Pull request by brenthagen**
# Overview
This adds the run log content to the run log tab on the protocol run details page by adapting the CommandList component and sub components from 5.0. "Download run log" is left as a TODO because it will require additional backend work, and the StepItem component will need additional styling updates.

closes #8753

# Changelog

 - Adds run log to protocol run details page

# Review requests

check run log before/during/after run. the new functionality is the "Jump to current step" button.

# Risk assessment

low

### CODE REVIEW

File: app/src/organisms/Devices/ProtocolRun/RunLogProtocolSetupInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+export const RunLogProtocolSetupInfo = ({
+  robotName,
+  runId,
+  setupCommand,
+}: RunLogProtocolSetupInfoProps): JSX.Element | null => {
+  const { t } = useTranslation('run_details')
+  const { protocolData } = useProtocolDetailsForRun(runId)
+  const protocolPipetteData = useRunPipetteInfoByMount(robotName, runId)
+
+  if (protocolData == null) return null
```
Comment (koji) [inline app/src/organisms/Devices/ProtocolRun/RunLogProtocolSetupInfo.tsx:30]: Using juggling-check is to catch `null` and `undefined`?

**Review by smb2268 — APPROVED** [review]
lots of pixel math in this one 😵‍💫 it looks good! I don't have a robot so wasn't able to test all of the possible run states but the basic functionality is working as expected on the simulator.

File: app/src/organisms/Devices/ProtocolRun/RunLogProtocolSetupInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        values={{
+          pipette_name: pipetteData.pipetteSpecs.displayName,
+          mount_name: setupCommand.params.mount === 'left' ? 'Left' : 'Right',
+        }}
+      />
+    )
+  } else if (setupCommand.commandType === 'loadModule') {
+    const moduleId = setupCommand.params.moduleId
+    const moduleModel = protocolData.modules[moduleId]
+    const moduleSlotNumber = moduleId.includes('thermocycler') ? 4 : 1
```
Comment (b-cooper) [inline app/src/organisms/Devices/ProtocolRun/RunLogProtocolSetupInfo.tsx:61]: Oooo we shouldn't be parsing the `moduleId`.  At the very least can we add a TODO comment here to parse this based on module model or type?

**Review by b-cooper — APPROVED** [review]
🏃 🪵

**Pull request merged**